code-challenges.test.js

```diff
@@ -15,27 +15,98 @@
+// Tests:    
+// Failed!!
+describe("deCode", () => {
+ 
+  it("takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.", () => {
+     const secretCodeWord1 = "Lackadaisical"
+    // Expected output: "L4ck4d41s1c4l"
+    const secretCodeWord2 = "Gobbledygook"
+    // Expected output: "G0bbl3dyg00k"
+    const secretCodeWord3 = "Eccentric"
+    // Expected output: "3cc3ntr1c" 
+    expect(deCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    expect(deCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    expect(deCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+  })
+})
+const deCode =(str)=>{
+    newStr1 =str.replaceAll(/a/gi,"4")
+    //converts a =>4
+    newStr2 =newStr1.replaceAll(/e/gi,"3")
+    //converts e =>3
+    newStr3 =newStr2.replaceAll(/i/gi, '1')
+    //converts i=>1
+    newStr4 =newStr3.replaceAll(/o/gi,"0")
+    //converts o=>0 &returns string
+    return newStr4
+}
+//str.replace(/([aeio])/g,'a':'4','e':'3', 'i':'1', 'o':'0')
+//str.replace(/a/,'4');
 
 // b) Create the function that makes the test pass.
 
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
+// ● wordLetter › akes in an array of words and a single letter and returns all the words that contain that particular 
+
+//     ReferenceError: wordLetter is not defined
+
+//       74 |     // Expected output: ["Cherry", "Blueberry", "Peach"]
+//       75 |     
+//     > 76 |     expect(wordLetter(rrayOfWords1,letterA)).toEqual(["Apple", "Banana", "Orange"])
+//          |     ^
+//       77 |     expect(wordLetter(arrayOfWords2,letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+//       78 |   })
+//       79 | })
+
+//       at Object.<anonymous> (code-challenges.test.js:76:5)
+
+// Test Suites: 1 failed, 1 total
+//Failed!
+describe("wordLetter", () => {
+ 
+  it("takes in an array of words and a single letter and returns all the words that contain that particular ", () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    const letterA = "a"
+    // Expected output: ["Apple", "Banana", "Orange"]
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    const letterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+    
+    expect(wordLetter(arrayOfWords1,letterA)).toEqual(["Apple", "Banana", "Orange"])
+    expect(wordLetter(arrayOfWords2,letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+  })
+})
+//create a function with params arr & search letter variable
+const wordLetter= (arr, query)=> {
+    //create a filter that searches array for a variable 
+    //returns corrected case to filter!
+  return arr.filter(function(el) {
+      //returns case to filter!
+    return el.toLowerCase().indexOf(query.toLowerCase()) !== -1
+  })
+}
```

Comment:
Logic looks good here.  Be sure to mention all of your built-in methods you are using in your pseudocode!  Also, it is best practice to make your parameters and placeholder variables descriptive so others can understand what is happening.  For instance, what does 'el' stand for in your filter?